File: FE/shopapp-angular/src/app/responses/prodducts/product.response.ts
export interface ProductResponse {
    id: number;
    name: string;
    price: number;
    image: string | null; // Image có thể là string (đường dẫn) hoặc null
    description: string;
    created_at: number[]; // Mảng chứa các giá trị số nguyên đại diện cho ngày tạo
    updated_at: number[]; // Mảng chứa các giá trị số nguyên đại diện cho ngày cập nhật
    category_id: number;
}

File: BE/shopapp/src/main/java/com/project/shopapp/DTOs/responses/ProductResponse.java
package com.project.shopapp.DTOs.responses;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.project.shopapp.models.Product;
import com.project.shopapp.models.ProductImage;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ProductResponse extends BaseResponse{
    private long id;
    private String name;
    private BigDecimal price;
    private String image;
    private String description;

    @JsonProperty("product_images")
    private List<ProductImageResponse> productImages;

    @JsonProperty("category_id")
    private Long categoryId;

//    @JsonProperty("name")
//    private String categoryName;

    public static ProductResponse fromProduct(Product product)
    {
        ProductResponse productResponse = ProductResponse.builder()
                .id(product.getId())
                .name(product.getName())
                .price(product.getPrice())
                .image(product.getImage())
                .description(product.getDescription())
                .categoryId(product.getCategory().getId())
                .build();
        productResponse.setCreatedAt(product.getCreatedAt());
        productResponse.setUpdatedAt(product.getUpdatedAt());
        return productResponse;
    }
}
